XRTK-Core/Packages/com.xrtk.core/Definitions/Controllers/Simulation/Hands/SimulatedHandControllerPoseData.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.Serialization;
+using XRTK.Definitions.Utilities;
+
+namespace XRTK.Definitions.Controllers.Simulation.Hands
+{
+    /// <summary>
+    /// A hand pose definition with recorded hand joint data.
+    /// Assign a collection of these poses to the data provider's profile to simulate the pose.
+    /// </summary>
+    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Input System/Controller Data Providers/Hands/Simulated Hand Controller Pose Data", fileName = "SimulatedHandControllerPoseData", order = (int)CreateProfileMenuItemIndices.Input)]
+    public class SimulatedHandControllerPoseData : BaseMixedRealityProfile
+    {
+        [SerializeField]
+        [Tooltip("Assign a unique ID to the pose. Can be used as a descriptive name, but must be unique!")]
+        private string id = string.Empty;
+
+        /// <summary>
+        /// A unique identifier for the hand pose.
+        /// </summary>
+        public string Id => id;
+
+        [SerializeField]
+        [Tooltip("Describe the hand pose.")]
+        [FormerlySerializedAs("gestureDescription")]
+        private string description = string.Empty;
+
+        /// <summary>
+        /// Gets the gesture's description.
+        /// </summary>
+        public string Description => description;
+
+        [SerializeField]
+        [Tooltip("Is this the default (idle) hand pose?")]
+        private bool isDefault = false;
+
+        /// <summary>
+        /// Gets whether this hand pose is the hand's idle pose.
+        /// </summary>
+        public bool IsDefault => isDefault;
+
+        [SerializeField]
+        [Tooltip("Key used to trigger the gesture simulation.")]
+        private KeyCode keyCode = KeyCode.None;
+
+        /// <summary>
+        /// Gets the key code used to trigger this gesture simulation.
+        /// </summary>
+        public KeyCode KeyCode => keyCode;
+
+        [SerializeField]
+        [Tooltip("Assign JSON definition file containing simulated gesture information.")]
+        private TextAsset data = null;
```

Comment:
I'm not sure if a TextAsset makes sense here.

Reply:
Why not? What would you prefer?

Reply:
Do TextAsset references survive builds? How does that get packed up? Maybe we want to serialize this as a string instead?

Reply:
Yes they survice builds. Don't see a string fit here because the TextAsset provided contains a lot of JSON that can be generated by recording a pose. And then for that JSON file you create a simulated pose defintion which makes the recorded psoe usable with the Simulation feature.

Reply:
Using a string requires pasting all the JSON into the serialized string field which makes the overall usablity of this feature a bit worse IMO